onnxruntime/core/providers/shared/initializer_view/initializer_view.cc

```diff
@@ -0,0 +1,24 @@
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/shared/initializer_view/initializer_view.h"
+
+namespace onnxruntime {
+std::unique_ptr<InitializerView> InitializerView::Create(
```

Comment:
>std::unique_ptr<InitializerView> [](http://example.com/codeflow?start=0&length=32)

Avoid dynamic memory allocations

Reply:
Good suggestion.  I have some choices here, 
- Use unique_ptr, which triggered dynamic memory allocations.
- Use Optional<T> as return value, which will touch a CopyConstructor.
- Set CTOR of InitializerView as public
- As @edgchen1   suggest,  pass a reference of Optional<T> as params in `Create`, would it be confusions of `_IN`, and `_OUT` semanteme?

Reply:
hm, I think a ctor might be simplest. unless there's a strong reason not to use exceptions?

Reply:
In almost all NNAPI/XNNPACK/COREML senario,   `UnpackInitializerData` is allowed to be failed.  If we throw a exception, every occurence of InitializerView should have `try-catch` around. It seems too expensive.